psd/imx6ull/psd.c

```diff
@@ -192,16 +191,32 @@ static int psd_controlBlock(uint32_t block)
 		err = fcb_flash(psd_common.f->oid, &psd_common.flash);
 	}
 	else if (block == DBBT) {
-		/* this should be done on whole chip or just kernel partitions */
-		printf("PSD: Check bad blocks - start: %u end: %llu.\n", 0, psd_common.flash.size / psd_common.flash.erasesz);
-		if ((err = flashmng_checkRange(psd_common.f->oid, 0, psd_common.flash.size, &psd_common.dbbt)) < 0)
+		if (psd_common.dbbt == NULL) {
+			psd_common.dbbt = calloc(1, sizeof(dbbt_t));
+			if (psd_common.dbbt == NULL) {
+				printf("PSD: Failed to alloc memory for DBBT.");
+				return -ENOMEM;
+			}
+		}
+
+		/* DBBT includes only bad blocks from kernel partitions */
+		printf("PSD: Check kernel1 bad blocks - start: %u end: %llu.\n", 0, psd_common.partsz / psd_common.flash.erasesz);
+		psd_changePartition(2);
```

Comment:
would be great if we wouldn't need to hardcode partition ids here (somehow triggered by PSD script?)